rllib/offline/estimators/tests/test_ope.py

```diff
@@ -199,22 +217,440 @@ def test_fqe_model(self):
         tmp_probs = []
         for act in range(fqe.policy.action_space.n):
             tmp_actions = np.zeros_like(self.batch["actions"]) + act
-            log_probs = fqe.policy.compute_log_likelihoods(
+            log_probs = self.policy.compute_log_likelihoods(
                 actions=tmp_actions,
                 obs_batch=self.batch["obs"],
             )
-            tmp_probs.append(torch.exp(log_probs))
-        tmp_probs = torch.stack(tmp_probs).transpose(0, 1)
-        tmp_probs = convert_to_numpy(tmp_probs)
+            tmp_probs.append(np.exp(log_probs))
+        tmp_probs = np.stack(tmp_probs).T
         check(action_probs, tmp_probs, decimals=3)
 
-    def test_multiple_inputs(self):
-        # TODO (Rohan138): Test with multiple input files
-        pass
+    def test_fqe_optimal_convergence(self):
+        # Optimal CliffWalkingWallPolicy with epsilon = 0.0
+        # and CliffWalkingWallEnv are deterministic;
+        # check that FQE converges to the true Q-values for self.batch
+        q_vals = [
+            -2.50,
+            -1.51,
+            -0.52,
+            0.49,
+            1.50,
+            2.53,
+            3.56,
+            4.61,
+            5.67,
+            6.73,
+            7.81,
+            8.90,
+            10.00,
+        ]
+        q_model_config = {
+            "tau": 1.0,
+            "model": {
+                "fcnet_hiddens": [],
+                "activation": "linear",
+            },
+            "lr": 0.01,
+            "n_iters": 5000,
+            "delta": 1e-3,
+        }
+
+        fqe = FQETorchModel(
+            policy=self.policy,
+            gamma=self.gamma,
+            **q_model_config,
+        )
+        losses = fqe.train(self.batch)
+        print(losses[-10:])
+        assert losses[-1] < fqe.delta, "FQE loss did not converge!"
```

Comment:
This loss_stop_condition  seems to be a bit aggressive. isn't it better to only check the estimates regardless of the final value of the loss? if the loss is not low enough it will show it in a non-converging behavior, right?

Reply:
It usually converges, but yeah, I can drop it